dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task05390RemoveEndpointIdForeignKeyInIntegrityResolverTablesIntegrationTest.java

```diff
@@ -0,0 +1,164 @@
+package com.dotmarketing.startup.runonce;
+
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.common.db.DotConnect;
+import com.dotmarketing.exception.DotDataException;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+
+
+public class Task05390RemoveEndpointIdForeignKeyInIntegrityResolverTablesIntegrationTest {
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+        // Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+    }
+
+    /**
+     * Method to Test: {@link Task05390RemoveEndpointIdForeignKeyInIntegrityResolverTables#executeUpgrade()}
+     * When: Run the Upgrade Task
+     * Should: Should remove the foreign key to all the Integrity Resolver tables and rename the column endpoint_id
+     * The Integrity Resolver tables are:
+     * - cms_roles_ir
+     * - folders_ir
+     * - structures_ir
+     * - htmlpages_ir
+     * - fileassets_ir
+     *
+     * After remove the constraint you should be allow to insert register in this table without has any register in publishing_end_point
+     */
+    @Test
+    public void constraintShouldNotExists() throws DotDataException {
+
+        final String endpointId = insertPublishingEndPoint();
+        insertFolderIntegrityResolver(endpointId);
+        insertPageIntegrityResolver(endpointId);
+        insertFileAssetIntegrityResolver(endpointId);
+        insertRolesIntegrityResolver(endpointId);
+        insertStructuresIntegrityResolver(endpointId);
+
+        final Task05390RemoveEndpointIdForeignKeyInIntegrityResolverTables task05390RemoveEndpointIdForeignKeyInIntegrityResolverTables =
+                new Task05390RemoveEndpointIdForeignKeyInIntegrityResolverTables();
+
+        task05390RemoveEndpointIdForeignKeyInIntegrityResolverTables.executeUpgrade();
+
+        insertFolderIntegrityResolver("anyIP");
```

Comment:
Codacy found an issue: [The String literal "anyIP" appears 5 times in this file; the first occurrence is on line 54](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6190239)